lib/iiif_print.rb

```diff
@@ -89,4 +90,43 @@ def iiif_print_config?
       end
     end
   end
+
+  # @api public
+  #
+  # Map the given model's metadata to the given IIIF version spec's metadata structure.  This
+  # is intended to be a drop-in replacement for `Hyrax::IiifManifestPresenter#manifest_metadata`.
+  #
+  # @param model [Object]
+  # @param version [Integer]
+  # @param fields [Array<IiifPrint::Metadata::Field>, Array<#name, #label>]
+  # @return [Array<Hash>]
+  #
+  # @see specs for expected output
+  #
+  # @see Hyrax::IiifManifestPresenter#manifest_metadata
+  def self.manifest_metadata_for(model:,
+                                 version: config.default_iiif_manifest_version,
+                                 fields: default_fields_for(model))
+    Metadata.manifest_for(model: model, version: version, fields: fields)
+  end
+
+  def self.metadata_fields
+    config.metadata_fields
+  end
+
+  # Hash is an arbitrary attribute key/value pairs
+  # Struct is a defined set of attribute "keys".  When we favor defined values,
+  # then we are naming the concept and defining the range of potential values.
+  Field = Struct.new(:name, :label, :options, keyword_init: true)
+
+  # TODO: figure out a way to use a custom label, right now it takes it get rendered from the title
+  def self.default_fields_for(_model, fields: metadata_fields)
+    fields.map do |field|
```

Comment:
Yowza… what is this "field" object we just got?  Why did it go from `field` to `field.first` and what is `field.last`?

I'm okay with no answer if we mark this method as `@api private`

However, it highlights that `metadata_fields` has shifted.  This is not a blocker but something to consider.

Reply:
I want to talk a bit more about this one as well.

Reply:
`fields` is our [hash](https://github.com/scientist-softserv/iiif_print/blob/55911876dd19ed45fcb7183896a4be64a454f9de/lib/iiif_print/configuration.rb#L27-L52)
`field` would be something like `[:creator, {:render_as=>:faceted}]`
`field.first` then is `:creator`
`field.last` is `{:render_as=>:faceted}`